File: level2/쇠막대기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <queue>

using namespace std;

int solution(string arrangement) {
    int answer = 0;
    int stick = 0;

    for(int i=0; i<arrangement.size(); i++){
        if(arrangement[i] == '(' && arrangement[i+1] == ')'){
            answer += stick;
            i++;
        }
        else if(arrangement[i] == '(')
            stick++;
        else if(arrangement[i] == ')'){
            stick--;
            answer += 1;
        }
    }
    
    return answer;
}

File: JAVA/level2/쇠막대기.java
class Solution {
    public int solution(String arrangement) {
        int answer = 0;
        int sticks = 0;

        for(int i=0; i<arrangement.length(); i++){
            if(arrangement.charAt(i) == '(' && arrangement.charAt(i+1) == ')'){
                answer += sticks;
                i++;
            }
            else if(arrangement.charAt(i) == '('){
                sticks++;
            }
            else{
                sticks--;
                answer ++;
            }
        }

        return answer;
    }
}
